services/outputhost/replicaconnection.go

```diff
@@ -186,124 +188,171 @@ func (conn *replicaConnection) readMessagesPump() {
 	// monotonically increasing
 	var lastSeqNum = int64(conn.startingSequence)
 
+	var skipOlderNanos = common.UnixNanoTime(conn.extCache.skipOlder * int32(time.Second))
```

Comment:
skipOlder should also respect the delay time, as discussed earlier. Just need to add the delay here.